graphrag/config/create_graphrag_config.py

```diff
@@ -375,6 +375,25 @@ def hydrate_parallelization_params(
                 container_name=reader.str(Fragment.container_name),
                 base_dir=reader.str(Fragment.base_dir) or defs.STORAGE_BASE_DIR,
             )
+
+        with (
+            reader.envvar_prefix(Section.update_index_storage),
+            reader.use(values.get("update_index_storage")),
+        ):
+            s_type = reader.str(Fragment.type)
+            if s_type:
+                update_index_storage_model = StorageConfig(
+                    type=StorageType(s_type) if s_type else defs.STORAGE_TYPE,
+                    connection_string=reader.str(Fragment.conn_string),
+                    storage_account_blob_url=reader.str(
+                        Fragment.storage_account_blob_url
+                    ),
+                    container_name=reader.str(Fragment.container_name),
+                    base_dir=reader.str(Fragment.base_dir)
+                    or defs.UPDATE_STORAGE_BASE_DIR,
+                )
+            else:
+                update_index_storage_model = None
```

Comment:
you could remove this else by moving the `update_index_storage_model = None` above the if :), could even go outside the with statement.

Reply:
I had it like that but the issue was that the interpreter was assuming this to always be set. So it greyed out some of my none checks and was driving me nuts.
I can try again by setting explicit typings.